### PR TITLE
Create Skills, Skills Category relationship table

### LINKED ISSUE #4691: Task - Create Skill, SkillCategory relationship table
Create many-many skill relationship table to associate skills with categories.
This means creating a data migration for the pivot table, and adding relationships to the model classes.

### PR DESCRIPTION
**Pull request by mnigh**
Resolves #4691

### Notes:
- Requires #4772 to be merged first.

### CODE REVIEW

**Review by tristan-orourke — COMMENTED** [review]
Looks good; waiting on the other PRs.

File: database/migrations/2020_11_24_150537_create_skill_categories_skills_table.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+class CreateSkillCategoriesSkillsTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('skill_categories_skills', function (Blueprint $table) {
```
Comment (tristan-orourke) [inline database/migrations/2020_11_24_150537_create_skill_categories_skills_table.php:16]: Accordion to Laravel [pivot table naming convention conenvtion](https://laravel.com/docs/8.x/eloquent-relationships#many-to-many-table-structure), this table should be called `skill_skill_category`.

File: app/Models/Skill.php [hunk 2]
```diff
@@ -95,6 +95,11 @@ public function experience_skills() // phpcs:ignore
         return $this->hasMany(\App\Models\ExperienceSkill::class);
     }
 
+    public function skill_categories() // phpcs:ignore
```
Comment (tristan-orourke) [inline app/Models/Skill.php:98]: Should add the relationship to the class docblock.

**Review by tristan-orourke — CHANGES_REQUESTED** [review]

**Comment by mnigh** [comment]
Code coverage test fails arbitrary coverage drop allowed for success value, 2.5%, by 0.06% (2.56%).

**Review by tristan-orourke — APPROVED** [review]

**Pull request merged**